owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java

```diff
@@ -0,0 +1,199 @@
+package edu.rice.owltorrent.network;
+
+import edu.rice.owltorrent.common.adapters.NetworkToStorageAdapter;
+import edu.rice.owltorrent.common.entity.FileBlockInfo;
+import edu.rice.owltorrent.common.entity.Peer;
+import edu.rice.owltorrent.common.entity.Torrent;
+import edu.rice.owltorrent.common.util.AtomicHashableInteger;
+import edu.rice.owltorrent.network.messages.PayloadlessMessage;
+import edu.rice.owltorrent.network.messages.PieceActionMessage;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Manager that takes care of everything related to a Torrent. The TorrentManager will manage the
+ * peer connections and ask for missing file pieces.
+ *
+ * @author Lorraine Lyu, Max Yu
+ */
+@Log4j2(topic = "general")
+public class TorrentManager implements Runnable, AutoCloseable {
+
+  private static final int DEFAULT_BLOCK_NUM = 2;
+
+  private static final int BLOCK_NOT_STARTED = 0;
+  private static final int BLOCK_IN_PROGRESS = 1;
+  private static final int BLOCK_DONE = 2;
+
+  private final Set<Integer> completedPieces;
+  private final Map<Integer, PieceStatus> uncompletedPieces;
+  private final Set<Integer> notStartedPieces;
+
+  private final int totalPieces;
+
+  private final Map<Peer, PeerConnector> peers;
+  private final NetworkToStorageAdapter networkStorageAdapter;
+  @Getter private final Torrent torrent;
+
+  public TorrentManager(Torrent file, NetworkToStorageAdapter adapter) {
+    this.torrent = file;
+    this.networkStorageAdapter = adapter;
+    this.completedPieces = Collections.newSetFromMap(new ConcurrentHashMap<>());
```

Comment:
These can just be set as fields, not in the constructor. Makes it cleaner imo

Reply:
So for the torrent we definitely don't want to make it a mutatable field, as the TorrentManager is designed to handle only one Torrent, and swapping the Torrent out would make things break. 
For the storage adapter changing it to a field would make sense when we want to change storages during runtime, but I don't think it's useful here?

Reply:
I meant the collections ones, like new set from map

Reply:
ohhhh i see what you mean now